packages/kbn-observable/src/operators/mergeMap.ts

```diff
@@ -40,49 +27,28 @@ export function mergeMap<T, I, R>(
  * that observable. You can think of it as being the same as `flatMap` on an
  * array, just that you return an Observable instead of an array.
  *
- * You can also specify a `resultSelector` that will receive both the value
- * emitted from the `source` observable and the value returned from the provided
- * `project` function, e.g.
+ * For more complex use-cases where you need the source variable for each value
+ * in the newly created observable, an often used pattern is using `map` within
+ * the `mergeMap`. E.g. let's say we want to return both the current value and
+ * the newly created value:
  *
  * ```js
- * mergeMap(
- *  pluginName => readPluginConfig$(pluginName),
- *  (pluginName, pluginConfig) => {
- *    return { pluginName, pluginConfig }
- *  }
+ * mergeMap(val =>
+ *   k$(someFn(val))(
+ *     map(newVal => ({ val, oldVal })
```

Comment:
__nit:__ `oldVal` --> `newVal`?

Reply:
Haha, I need tests for my examples 🙈

Reply:
Heh, we desperately need [`rustdoc`](https://doc.rust-lang.org/stable/rustdoc/documentation-tests.html) analog for JS that compiles and runs examples from markdown comments as real tests :) Or maybe there is something already that I don't know about...

Reply:
That would be so cool!